src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -163,25 +163,40 @@ void Qgs3DMapCanvas::setMapTool( Qgs3DMapTool *tool )
 
 bool Qgs3DMapCanvas::eventFilter( QObject *watched, QEvent *event )
 {
-  if ( !mMapTool )
-    return false;
+  if ( mMapTool && watched == mapTool() )
+  {
+    switch ( event->type() )
+    {
+      case QEvent::MouseButtonPress:
+        mMapTool->mousePressEvent( static_cast<QMouseEvent *>( event ) );
+        break;
+      case QEvent::MouseButtonRelease:
+        mMapTool->mouseReleaseEvent( static_cast<QMouseEvent *>( event ) );
+        break;
+      case QEvent::MouseMove:
+        mMapTool->mouseMoveEvent( static_cast<QMouseEvent *>( event ) );
+        break;
+      default:
+        break;
+    }
+  }
 
-  Q_UNUSED( watched )
-  switch ( event->type() )
+  if ( mContainer && watched == mContainer )
   {
-    case QEvent::MouseButtonPress:
-      mMapTool->mousePressEvent( static_cast<QMouseEvent *>( event ) );
-      break;
-    case QEvent::MouseButtonRelease:
-      mMapTool->mouseReleaseEvent( static_cast<QMouseEvent *>( event ) );
-      break;
-    case QEvent::MouseMove:
-      mMapTool->mouseMoveEvent( static_cast<QMouseEvent *>( event ) );
-      break;
-    default:
-      break;
+    if ( event->type() == QEvent::Resize )
+      updateViewportSize();
```

Comment:
Are the changes in this file still necessary? Or can we roll them back?

Reply:
sure!